jwt/api.py

```diff
@@ -110,14 +129,16 @@ def encode(self, payload, key, algorithm='HS256', headers=None, json_encoder=Non
 
         return b'.'.join(segments)
 
-    def decode(self, jwt, key='', verify=True, algorithms=None, **kwargs):
+    def decode(self, jwt, key='', verify=True, algorithms=None, options=None, **kwargs):
         payload, signing_input, header, signature = self._load(jwt)
 
         if verify:
-            self._verify_signature(payload, signing_input, header, signature,
-                                   key, algorithms)
+            merged_options = self._merge_options(override_options=options)
+            if merged_options.get('verify_signature'):
+                self._verify_signature(payload, signing_input, header, signature,
+                                       key, algorithms)
```

Comment:
How about adding common non-standard claims verification to the api? This way you have no single highly complex interface. Adding an options keyword argument that can take five different keys is the same as adding those keys to the method interface:

    decode(self, jwt, key='', verify=True, algorithms=None, verify_signature=False, verify_exp=False, verify_nbf=False, verify_iat=False, verify_aud=False, **kwargs)

The order of complexity is increased for each option.

As experts in the use cases, you will give less experienced users better information about how they should do things.